storage/innobase/handler/ha_innodb.cc

```diff
@@ -12273,6 +12279,9 @@ create_table_info_t::create_option_encryption_is_valid() const
 	ulint		fsp_flags = space->flags;
 
 	bool tablespace_is_encrypted = FSP_FLAGS_GET_ENCRYPTION(fsp_flags);
+	if (fsp_is_system_temporary(space_id)) {
```

Comment:
Why can't we have proper fsp_flags for temp tablespace? then we don't need this special condition

Reply:
Then we will need similar condition in other places, when we trying to from temporary tablespace, because pages may be or may be not encrypted regardless of the tablespace flag.

Reply:
Ok, the core change to make write to temp tablespace unencrypted is already there before your PR. 

fil_io_set_encryption()
```
  /* For writing temporary tablespace, if encryption for temporary
        tablespace is disabled, skip setting encryption. */
        if (fsp_is_system_temporary(space->id)
            && !srv_tmp_tablespace_encrypt && req_type.is_write()) {
                req_type.clear_encrypted();
                return;
        }
``
The writes are already handled(see above) and this is the main change to allow table creation?

Reply:
Yep, writes were already handled in the `fil_io_set_encryption` and it was actually a bug, because it did not match the documented behaviour. I thought it was obvious and didn't mention. Now we turned it into a feature. And this PR only adds handling of CREATE TABLE statement.

Reply:
It is a "hidden feature" :laughing: